src/math.rs

```diff
@@ -10,6 +10,17 @@ pub fn lerp(first: &f32, second: &f32, numerator: u32, denominator: u32) -> f32
     first + (second - first) * numerator as f32 / denominator as f32
 }
 
+/// short macro to generate a `NonZero`. It panics during compile if the
+/// passed in literal is zero. Used for `ChannelCount` and `Samplerate`
+/// constants
+macro_rules! nz {
```

Comment:
A similar macro `nonzero` is provided by [nonzero_ext](https://crates.io/crates/nonzero_ext). I understand it may be too heavy-weight to pull in a dependency just for that, but maybe we could use the same name for recognizability and ease of switching if we ever want to use more of `nonzero_ext` in the future.

Reply:
I had a look its not a transitive dependencies of rodio so for now lets keep it out. (Another argument is that `nz!` is nice in short, though that might be bad taste on my end). 

If you ever need something from nonzero_ext do not hesitate to include it though!